crates/ironrdp-connector/src/connection.rs

```diff
@@ -253,13 +178,15 @@ impl ClientConnector {
         debug_assert!(!self.should_perform_security_upgrade());
     }
 
-    pub fn is_credssp_step(&self) -> bool {
-        matches!(
-            &self.state,
-            ClientConnectorState::CredsspInitial { .. }
-                | ClientConnectorState::CredsspReplyNeeded { .. }
-                | ClientConnectorState::CredsspEarlyUserAuthResult { .. }
-        )
+    pub fn should_perform_credssp(&self) -> bool {
+        matches!(self.state, ClientConnectorState::CredSsp { .. })
+    }
+
+    pub fn mark_credssp_as_done(&mut self) -> Written {
+        assert!(self.should_perform_credssp());
+        let res = self.step(&[], &mut WriteBuf::new()).expect("transition to next state");
+        debug_assert!(!self.should_perform_credssp());
+        res
```

Comment:
This function should not return a `Written`, because we don’t expect anything from the connector in addition to the state transition. However, as a sanity check, you can assert for `Written::Nothing` if you want.
```suggestion
    pub fn mark_credssp_as_done(&mut self) {
        assert!(self.should_perform_credssp());
        let res = self.step(&[], &mut WriteBuf::new()).expect("transition to next state");
        assert_eq!(res, Written::Nothing);
        debug_assert!(!self.should_perform_credssp());
```